src/main/java/seedu/tatoolkit/ui/MainWindow.java

```diff
@@ -187,8 +187,13 @@ private CommandResult executeCommand(String commandText) throws CommandException
             logger.info("Result: " + commandResult.getFeedbackToUser());
             resultDisplay.setFeedbackToUser(commandResult.getFeedbackToUser());
 
-            Optional<Person> lastViewedPerson = logic.getLastViewedPerson();
-            lastViewedPerson.ifPresentOrElse(sidePanel::displayPerson, sidePanel::resetDetails);
+            logic.getLastExecutedCommand().ifPresent(command -> {
+                if (command instanceof ViewCommand) {
+                    logic.getLastViewedPerson().ifPresentOrElse(sidePanel::displayPerson, sidePanel::resetDetails);
```

Comment:
A bit unsure of how this works - so if there is a last viewed person then we display, or else we reset the side panel? Correct me if I'm wrong, but if the ViewCommand exists already then there must be a last viewed person right. My bad if I misunderstood something, just wanted to clarify how this works!

Reply:
You are probably right to assume that LastViewPerson should be present if the ViewCommand was called previously. However, I think I will just leave the check there in the spirit of defensive programming and the use of Optional to explicitly handle the absence of values.

Reply:
Very exciting! I love to see practical implementations of theoretical concepts that we have studied in our CS2103T weekly lectures. Godpseed to you, good friend.

Reply:
Godspeed brother.